app/controllers/like_controller.rb

```diff
@@ -0,0 +1,19 @@
+class LikesController < ApplicationController
+  def new
+    @like = Like.new
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:post_id])
+    @like = @post.likes.build(user: current_user)
+
+    if @like.save
+      redirect_to user_post_path(@user, @post), notice: 'Like was successfully created.'
+    else
+      render :new
+    end
+  end
+end
```

Comment:
- [ ] [REQUIRED]  I noticed that you currently have a new action along with the create action. While having a new action is not strictly necessary for creating likes, it seems like you might be implementing a form-based approach.

For a more streamlined implementation, you can consider simplifying your controller to only include the create action. Here's an example of how you can modify it:
``` ruby
class LikesController < ApplicationController
  def create
    @post = Post.find(params[:post_id])
    @like = @current_user.likes.new(post: @post)

    respond_to do |format|
      if @like.save
        format.html { redirect_to user_post_path(@post.author, @post), notice: 'Like was successfully created.' }
      else
        format.html { redirect_to user_post_path(@post.author, @post), alert: 'Failed to create like.' }
      end
    end
  end
end
```

- NB: I have refactored the code to remove the new action and the associated view. If the like creation is successful, the user is redirected to the post's path with a success notice. If the creation fails, they are redirected with an alert.